aten/src/ATen/native/Histogram.cpp

```diff
@@ -74,13 +74,13 @@ void histogramdd_check_inputs(const Tensor& input, const TensorList& bins, const
     TORCH_CHECK(input.dim() >= 2, "torch.histogramdd: input tensor should have at least 2 dimensions, but got ",
                 input.dim());
 
-    const int64_t N = input.size(-1);
+    const SymInt N = input.sym_size(-1);
 
     TORCH_CHECK(static_cast<int64_t>(bins.size()) == N, "torch.histogramdd: expected ", N, " sequences of bin edges for a ", N,
                 "-dimensional histogram but got ", bins.size());
 
     auto input_dtype = input.dtype();
-    for (const auto dim : c10::irange(N)) {
+    for (auto dim = 0; dim < N; ++dim) {
```

Comment:
`irange` doesn't support `SymInt`.

Reply:
This doesn't make sense. To iterate over N, you need to know concretely what it is, so that you know how many iterations are. So this is essentially equivalent to having guarded on N when you extracted it out of input. It's reasonable to only support symbolic sizes on some dims of inputs; I recommend just guarding.